pkg/plugins/runtime/k8s/controllers/pod_converter_test.go

```diff
@@ -885,6 +903,85 @@ var _ = Describe("ProtocolTagFor(..)", func() {
 	)
 })
 
+var _ = Describe("Serviceless Name for(...)", func() {
+	type testCase struct {
+		pod          string
+		replicaSets  string
+		jobs         string
+		expectedName string
+		expectedKind string
+	}
+	DescribeTable("should infer name based on the resource type",
+		func(given testCase) {
+			// given
+			ctx := context.Background()
+
+			pod := &kube_core.Pod{}
+			bytes, err := os.ReadFile(filepath.Join("testdata", "serviceless", given.pod))
+			Expect(err).ToNot(HaveOccurred())
+			err = yaml.Unmarshal(bytes, pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			var replicaSetGetter kube_client.Reader
+			if given.replicaSets != "" {
+				replicaSetGetter = getReplicaSetsReader("testdata", "serviceless", given.replicaSets)
+			}
+
+			var jobGetter kube_client.Reader
+			if given.jobs != "" {
+				jobGetter = getJobsReader("testdata", "serviceless", given.jobs)
+			}
+
+			nameExtractor := NameExtractor{
+				ReplicaSetGetter: replicaSetGetter,
+				JobGetter:        jobGetter,
+			}
+
+			// when
+			name, kind, err := nameExtractor.Name(ctx, pod)
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+			Expect(name).To(Equal(given.expectedName))
+			Expect(kind).To(Equal(given.expectedKind))
+		},
+		Entry("name from deployment", testCase{
+			pod:          "01.pod.yaml",
+			replicaSets:  "01.replicasets-for-pod.yaml",
+			expectedName: "test-server",
+			expectedKind: "Deployment",
+		}),
+		Entry("name from cronjob", testCase{
+			pod:          "02.pod.yaml",
+			jobs:         "02.job-for-pod.yaml",
+			expectedName: "test-job",
+			expectedKind: "CronJob",
+		}),
+		Entry("name from replicaset", testCase{
+			pod:          "03.pod.yaml",
+			replicaSets:  "03.replicasets-for-pod.yaml",
+			expectedName: "test",
```

Comment:
I think we should fix this but this is *breaking change*

Reply:
I for one am for breaking it